src/memory/primitives/relocatable.test.ts

```diff
@@ -0,0 +1,99 @@
+import { test, expect, describe } from 'bun:test';
+import { Felt } from './felt';
+import {
+  Relocatable,
+  OffsetUnderflow,
+  SegmentError,
+  ForbiddenOperation,
+} from './relocatable';
+import { UnsignedInteger } from './uint';
+
+describe('Relocatable', () => {
+  describe('constructor', () => {
+    test('should initialise a relocatable correctly', () => {
+      const relocatable = new Relocatable(0, 5);
+      expect(relocatable.getSegmentIndex()).toEqual(0);
+      expect(relocatable.getOffset()).toEqual(5);
+    });
+  });
+
+  describe('sub', () => {
+    test('should subtract two relocatables properly within the same segment', () => {
+      const a = new Relocatable(1, 10);
+      const b = new Relocatable(1, 3);
+      const result = a.sub(b);
+      expect(result.getOffset()).toEqual(7);
+    });
+
+    test('should throw OffsetUnderflow when offset goes below zero', () => {
+      const a = new Relocatable(1, 2);
+      const b = new Relocatable(1, 3);
+      expect(() => a.sub(b)).toThrow(new OffsetUnderflow());
+    });
+
+    test('should throw SegmentError when segments are different', () => {
+      const a = new Relocatable(1, 10);
+      const b = new Relocatable(2, 5);
+      expect(() => a.sub(b)).toThrow(new SegmentError());
+    });
+    test('should subtract a Felt from a Relocatable', () => {
+      const relocatable = new Relocatable(0, 10);
+      const felt = new Felt(5n);
+      const result = relocatable.sub(felt);
+      expect(result.getOffset()).toEqual(5);
+    });
+
+    test('should throw OffsetUnderflow when subtracting a larger Felt', () => {
+      const relocatable = new Relocatable(0, 5);
+      const felt = new Felt(10n);
+      expect(() => relocatable.sub(felt)).toThrow(new OffsetUnderflow());
```

Comment:
in `uint.test.ts` you are using this syntax for checking that a test throws:
```
expect(() => UnsignedInteger.toUint(-5)).toThrow(TypeError);
```
can we sync on one syntax?